language-server/src/jsonc-instance.js

```diff
@@ -196,6 +196,41 @@ export class JsoncInstance {
   textLength() {
     return this.node.length;
   }
+
+  keyAtPosition(position) {
+    const positionOffset = this.textDocument.offsetAt(position);
+
+    if (this.typeOf() !== "object") {
+      return undefined;
+    }
+
+    for (const propertyNode of this.node.children) {
+      const propertyNameNode = propertyNode.children[0];
+      const propertyStartOffset = propertyNameNode.offset;
+      const propertyEndOffset = propertyStartOffset + propertyNameNode.length;
+
+      if (positionOffset >= propertyStartOffset && positionOffset <= propertyEndOffset) {
+        let pointer;
+        if (this.pointer === "") {
+          pointer = `/${propertyNameNode.value}`;
+        } else {
+          pointer = `${this.pointer}/${propertyNameNode.value}`;
+        }
+
+        return new JsoncInstance(this.textDocument, this.root, propertyNameNode, pointer, this.annotations);
+      }
+
+      if (propertyNameNode.type === "object") {
+        const nestedInstance = new JsoncInstance(this.textDocument, this.root, propertyNameNode, this.pointer, this.annotations);
+        const nestedKey = nestedInstance.keyAtPosition(position);
+        if (nestedKey !== undefined) {
+          return nestedKey;
+        }
+      }
+    }
+
+    return undefined;
```

Comment:
Don't return `undefined` if the location isn't found. In these cases, I've settled on the pattern of returning a JsoncInstance that represents an undefined value. See the `get` method for an example.